internal/services/securitycenter/security_center_storage_defender_resource.go

```diff
@@ -251,6 +265,7 @@ func (s StorageDefenderResource) Read() sdk.ResourceFunc {
 							state.MalwareScanningOnUploadEnabled = pointer.From(onUpload.IsEnabled)
 							state.MalwareScanningOnUploadCapPerMon = pointer.From(onUpload.CapGBPerMonth)
 						}
+						state.ScanResultsEventGridTopicId = pointer.From(ms.ScanResultsEventGridTopicResourceId)
```

Comment:
We usually parse resource IDs that are returned by Azure before setting into state to ensure the casing is correct (or to fix the casing if it's returned incorrectly)

```suggestion
						if ms.ScanResultsEventGridTopicResourceId != nil {
							topicId, err := topics.ParseTopicID(*ms.ScanResultsEventGridTopicResourceId)
							if err != nil {
								return err
							}
							state.ScanResultsEventGridTopicId = topicId.ID()
```
						}